celery_serverless/invoker.py

```diff
@@ -219,3 +238,57 @@ def invoke(target='watchdog', config=None, *args, **kwargs):
 
 def invoke_worker(config=None, data=None, *args, **kwargs):
     return invoke(target='worker', extra_data=data or {}, *args, **kwargs)
+
+
+def invoke_watchdog(config=None, data=None, check_lock=True, *args, **kwargs):
+    from .watchdog import _get_watchdog_lock
+
+    if check_lock:
+        lock_watchdog = _get_watchdog_lock(enforce=False)[0]
+        if lock_watchdog.locked():
+            logger.info('Watchdog lock already held. Giving up.')
+            return False, RuntimeError('Watchdog lock already held')
+    return invoke(target='watchdog', extra_data=data or {}, *args, **kwargs)
+
+
+def client_invoke_watchdog(check_lock=True, blocking_lock=False, *args, **kwargs):
+    if not check_lock:
+        logger.debug('Not checking client lock')
+        return invoke_watchdog(check_lock=True, *args, **kwargs)
+
+    client_lock = _get_client_lock()[0]
+    locked = client_lock.acquire(blocking_lock)
+    if not locked:
+        logger.info('Could not get Client lock. Giving up.')
+        return False, RuntimeError('Client lock already held')
+
+    logger.debug('Got the client lock')
+    try:
+        return invoke_watchdog(with_lock=True, *args, **kwargs)
+    finally:
+        logger.debug('Releasing the client lock')
+        client_lock.release()
```

Comment:
Since a `invoke_watchdog()` was just made, we could let the `client_lock` expire by itself as long as it expires before the lifespan of a watchdog. This could avoid an eventual call by another client. Only works for distributed locks that expire though.

Reply:
Yes. One though is to put a timeout instead of unlock the client_lock if the watchdog got invoked with success.

But right now the client lock can be a `multiprocessing.Lock` if Redis is not available on the client, or is undesirable to flood Redis with lock checks.

What do you think? Do this now or let go for some other PR?

Reply:
I think it is ok as it is for now, lets just keep it in mind for future work.

Reply:
Issue created for future work: #22